src/static/js/cockpit.js

```diff
@@ -11,21 +11,47 @@
 
   //Cockpit is inheriting from EventEmitter2.
   var Cockpit = function Cockpit(csocket) {
-    this.socket = csocket;
+    var self = this;
+    this.uiLoader = new window.UiLoader();
+    this.rov = new window.MessageManager(csocket);
     this.sendUpdateEnabled = true;
     this.capabilities = 0;
     this.loadedPlugins = [];
-    this.loadPlugins();
-    console.log('loaded plugins');
-    // Register the various event handlers
-    this.listen();
+
+    this.loadUi(function() {
+      self.extensionPoints = {
+        rovSettings: $('html /deep/ rov-settings'),
+        rovDiagnostics: $('html /deep/ rov-diagnostics /deep/ #dropIn'),
+        videoContainer: $('html /deep/ rov-video'),
+        keyboardInstructions: $('html /deep/ #keyboardInstructions'),
+        buttonPanel: $('html /deep/ #buttonPanel'),
+        menu: $('html /deep/ rov-menu'),
+        inputController: undefined //will be set by the plugin
```

Comment:
The input controller as an extensionpoint seems at fist glance to break the mental model of how extensionpoints should work.  Where as the rest of the extension points expose a inner DOM for adding elements, the input control features an API for registering controls.  While I get the concept of your "extending" the control model, it seems like a very different type of extension and one what should live outside of the extensionpoints.